docs/e2e-testing.md

```diff
@@ -0,0 +1,139 @@
+
+### Configuring browsers for E2E Testing
+
+TEAMMATES E2E testing requires Firefox or Chrome.
+
+Before running tests, modify `src/e2e/resources/test.properties` if necessary, e.g. to configure which browser and test accounts to use.
+
+#### Using Firefox
+
+* You need to use geckodriver for testing with Firefox.
+  * Download the latest stable geckodriver from [here](https://github.com/mozilla/geckodriver/releases).
+    The site will also inform the versions of Firefox that can be used with the driver.
+  * Specify the path to the geckodriver executable in `test.geckodriver.path` value in `test.properties`.
+
+* If you want to use a Firefox version other than your computer's default, specify the custom path in `test.firefox.path` value in `test.properties`.
+
+* If you are planning to test against a production server, specify the Firefox profile to be used in `test.firefox.profile.name` value in `test.properties`.
+    *  This is used to bypass login by using previous login data.
+    *  You can enter `about:profiles` into Firefox address bar to identify the profile being used.
+
+#### Using Chrome
+
+* You need to use chromedriver for testing with Chrome.
+  * Download the latest stable chromedriver from [here](https://sites.google.com/a/chromium.org/chromedriver/downloads).
+    The site will also inform the versions of Chrome that can be used with the driver.
+  * Specify the path to the chromedriver executable in `test.chromedriver.path` value in `test.properties`.
+
+* If you are planning to test against a production server, specify the path to Chrome's user data directory in `test.chrome.userdata.path` value in `test.properties`.
+    *  This is used to bypass login by using previous login data.
+
+* The chromedriver process started by the test suite will not automatically get killed after the tests have finished executing.<br>
+  You will need to manually kill these processes after the tests are done.
+  * On Windows, use the Task Manager or `taskkill /f /im chromedriver.exe` command.
+  * On OS X, use the Activity Monitor or `sudo killall chromedriver` command.
+
+### Running the tests
+- When running the test cases, a few cases may fail (this can happen due to timing issues). They can be re-run until they pass without affecting the accuracy of the tests.
+
+#### Running the tests with command line
+E2E tests follow this configuration:
+
+Test suite | Command | Results can be viewed in
+---|---|---
+`E2E tests` | `./gradlew e2eTests` | `{project folder}/build/reports/e2e-test-try-{n}/index.html`, where `{n}` is the sequence number of the test run
+Any individual E2E test | `./gradlew e2eTestTry1 --tests TestClassName` | `{project folder}/build/reports/e2e-test-try-1/index.html`
+
+
+- `E2E tests` will be run in their entirety once and the failed tests will be re-run a few times. 
+- Before running `E2E tests`, it is important to have the both front-end and back-end dev servers running locally first if you are testing against them.
```

Comment:
It is mentioned like this before, but this part is not 100% accurate. Only the back-end dev server is absolutely necessary, in particular if the front-end has been built. I suggest to shorten the term to just "dev server"; anyone who is advanced enough to run E2E tests should be able to figure this out on their own.

it is important to have the ~both front-end and back-end~ dev server~s~ running locally first if you are testing against ~them~ it.